File: server/aoNewsletter2/Models/Db/NewsletterModel.vb
Imports System
Imports Contensive.Addons.Newsletter.Models.Db
Imports Contensive.BaseClasses
Imports Contensive.Models.Db

Namespace Models.Db
    Public Class NewsletterModel
        Inherits DesignBlockBaseModel

        Public Shared ReadOnly Property tableMetadata As DbBaseTableMetadataModel = New DbBaseTableMetadataModel("Newsletters", "Newsletters", "default", False)
        '
        Public Property templateId As Integer
        Public Property stylesFileName As DbBaseModel.FieldTypeCSSFile
        Public Property emailTemplateId As Integer
        Public Property mastheadFilename As String
        Public Property footerFilename As String
        Public Property blockArchiveSearchForm As Boolean
        Public Property archiveIssuesToDisplay As Integer
        Public Property searchResultsPerPage As Integer

        Public Shared Function createOrAddSettings(ByVal cp As CPBaseClass, ByVal settingsGuid As String) As NewsletterModel
            Dim result As NewsletterModel = create(Of NewsletterModel)(cp, settingsGuid)

            If (result Is Nothing) Then
                result = NewsletterModel.addDefault(Of NewsletterModel)(cp)
                result.name = NewsletterModel.tableMetadata.contentName & " " & result.id
                result.ccguid = settingsGuid
                result.themeStyleId = 0
                result.padTop = False
                result.padBottom = False
                result.padRight = False
                result.padLeft = False

                result.save(cp)
                cp.Content.LatestContentModifiedDate.Track(result.modifiedDate)
            End If

            Return result
        End Function
    End Class
End Namespace

File: server/aoNewsletter2/Models/Db/NewsletterModel.cs
using Contensive.BaseClasses;
using Contensive.Models.Db;

namespace Contensive.Addons.Newsletter.Models.Db {
    public class NewsletterModel : DesignBlockBaseModel {

        public static DbBaseTableMetadataModel tableMetadata { get; private set; } = new DbBaseTableMetadataModel("Newsletters", "Newsletters", "default", false);
        // 
        public int templateId { get; set; }
        public FieldTypeCSSFile stylesFileName { get; set; }
        public int emailTemplateId { get; set; }
        public string mastheadFilename { get; set; }
        public string footerFilename { get; set; }
        public bool blockArchiveSearchForm { get; set; }
        public int archiveIssuesToDisplay { get; set; }
        public int searchResultsPerPage { get; set; }

        public static NewsletterModel createOrAddSettings(CPBaseClass cp, string settingsGuid) {
            var result = create<NewsletterModel>(cp, settingsGuid);

            if (result is null) {
                result = addDefault<NewsletterModel>(cp);
                result.name = tableMetadata.contentName + " " + result.id;
                result.ccguid = settingsGuid;
                result.themeStyleId = 0;
                result.padTop = false;
                result.padBottom = false;
                result.padRight = false;
                result.padLeft = false;

                result.save(cp);
                cp.Content.LatestContentModifiedDate.Track(result.modifiedDate);
            }

            return result;
        }
    }
}